go/cmd/dolt/commands/schcmds/import.go

```diff
@@ -16,26 +16,313 @@ package schcmds
+			err := filesys.UnmarshalJSONFile(dEnv.FS, mappingFile, &m)
+
+			if err != nil {
+				return errhand.BuildDError("error: invalid mapper file.").AddCause(err).Build()
+			}
+
+			colMapper = actions.MapMapper(m)
+		} else {
+			return errhand.BuildDError("error: '%s' does not exist.", mappingFile).Build()
+		}
+	} else {
+		colMapper = actions.IdentityMapper{}
+	}
+
+	floatThresholdStr := apr.GetValueOrDefault(floatThresholdParam, "0.0")
+	floatThreshold, err := strconv.ParseFloat(floatThresholdStr, 64)
+
+	if err != nil {
+		return errhand.BuildDError("error: '%s' is not a valid float in the range 0.0 (all floats) to 1.0 (no floats)", floatThresholdStr).SetPrintUsage().Build()
+	}
+
+	delim := apr.GetValueOrDefault(delimParam, ",")
+
+	impArgs := importArgs{
+		op:       op,
+		fileName: fileName,
+		delim:    delim,
+		fileType: apr.GetValueOrDefault(fileTypeParam, filepath.Ext(fileName)),
+		inferArgs: &actions.InferenceArgs{
+			ExistingSch:    existingSch,
+			ColMapper:      colMapper,
+			FloatThreshold: floatThreshold,
+			KeepTypes:      apr.Contains(keepTypesParam),
+		},
+	}
+
+	sch, verr := inferSchemaFromFile(ctx, dEnv.DoltDB.ValueReadWriter().Format(), pks, &impArgs)
+
+	if verr != nil {
+		return verr
+	}
+
+	cli.Println(sql.SchemaAsCreateStmt(tblName, sch))
+
+	if !apr.Contains(dryRunFlag) {
+		schVal, err := encoding.MarshalAsNomsValue(context.Background(), root.VRW(), sch)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to encode schema.").AddCause(err).Build()
+		}
+
+		m, err := types.NewMap(ctx, root.VRW())
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to create table.").AddCause(err).Build()
+		}
+
+		tbl, err = doltdb.NewTable(ctx, root.VRW(), schVal, m)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to create table.").AddCause(err).Build()
+		}
+
+		root, err = root.PutTable(ctx, tblName, tbl)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to add table.").AddCause(err).Build()
+		}
+
+		err = dEnv.UpdateWorkingRoot(ctx, root)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to update the working set.").AddCause(err).Build()
+		}
+
+		cli.PrintErrln(color.CyanString("Created table successfully."))
+	}
+
+	return nil
+}
+
+func inferSchemaFromFile(ctx context.Context, nbf *types.NomsBinFormat, pkCols []string, args *importArgs) (schema.Schema, errhand.VerboseError) {
+	if args.fileType[0] == '.' {
+		args.fileType = args.fileType[1:]
+	}
+
+	var rd table.TableReadCloser
+	switch args.fileType {
+	case "csv":
+		f, err := os.Open(args.fileName)
+
+		if err != nil {
+			return nil, errhand.BuildDError("error: failed to open '%s'", args.fileName).Build()
+		}
+
+		defer f.Close()
+
+		rd, err = csv.NewCSVReader(nbf, f, csv.NewCSVInfo().SetDelim(args.delim))
+
+		if err != nil {
+			return nil, errhand.BuildDError("error: failed to create a CSVReader.").AddCause(err).Build()
+		}
+
+		defer rd.Close(ctx)
+	}
```

Comment:
Probably should throw an error here on unsupported file types.

Reply:
fixed